packages/application-generic/src/usecases/conditions-filter/conditions-filter.usecase.ts

```diff
@@ -461,11 +511,98 @@ export class ConditionsFilter extends Filter {
     command: ConditionsFilterCommand,
     filterProcessingDetails: FilterProcessingDetails
   ): Promise<boolean> {
-    const foundFilter = await this.findAsync(filter.children, (i) =>
+    const { webhookFilters, otherFilters } = this.splitFilters(filter);
+
+    const foundFilter = await this.findAsync(otherFilters, (i) =>
       this.processFilter(variables, i, command, filterProcessingDetails)
     );
+    if (foundFilter) {
+      return true;
+    }
 
-    return !!foundFilter;
+    return !!(await this.findAsync(webhookFilters, (i) =>
+      this.processFilter(variables, i, command, filterProcessingDetails)
+    ));
+  }
+
+  @Instrument()
+  private async normalizeVariablesData(command: ConditionsFilterCommand) {
+    const filterVariables: IFilterVariables = {};
+
+    filterVariables.subscriber = await this.fetchSubscriberIfMissing(command);
+    filterVariables.tenant = await this.fetchTenantIfMissing(command);
+    filterVariables.payload = command.variables?.payload
+      ? command.variables?.payload
+      : command.job?.payload ?? undefined;
+
+    return filterVariables;
+  }
+
+  private async fetchSubscriberIfMissing(
+    command: ConditionsFilterCommand
+  ): Promise<SubscriberEntity | undefined> {
+    if (command.variables?.subscriber) {
+      return command.variables.subscriber;
+    }
+
+    const subscriberFilterExist = command.step?.filters?.find((filter) => {
+      return filter?.children?.find((item) => item?.on === 'subscriber');
+    });
```

Comment:
@djabarovgeorge what would happen here if there is a condition on subscriber in a variant? Or in the future with integration?
Could we make this more general?

Reply:
This state is not possible because we made select variant command with a mandatory variable, however, due to the reasoning we could have other clients of condition filters in the future i added the addition logit in order to fetch the possible condition from variants.